src/themes.rs

```diff
@@ -166,51 +169,123 @@ lazy_static! {
     };
 }
 
-mapped_struct! {
-    #[derive(Deserialize, Debug, Default, Clone)]
-    #[serde(deny_unknown_fields)]
-    pub struct Theme: String {
-        pub idle_bg,
-        pub idle_fg,
-        pub info_bg,
-        pub info_fg,
-        pub good_bg,
-        pub good_fg,
-        pub warning_bg,
-        pub warning_fg,
-        pub critical_bg,
-        pub critical_fg,
-        pub separator,
-        pub separator_bg,
-        pub separator_fg,
-        pub alternating_tint_bg,
-        pub alternating_tint_fg
+#[derive(Deserialize, Debug, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Theme {
+    pub idle_bg: String,
+    pub idle_fg: String,
+    pub info_bg: String,
+    pub info_fg: String,
+    pub good_bg: String,
+    pub good_fg: String,
+    pub warning_bg: String,
+    pub warning_fg: String,
+    pub critical_bg: String,
+    pub critical_fg: String,
+    pub separator: String,
+    pub separator_bg: String,
+    pub separator_fg: String,
+    pub alternating_tint_bg: String,
+    pub alternating_tint_fg: String,
+}
+
+impl Default for Theme {
+    fn default() -> Self {
+        PLAIN.clone()
     }
 }
 
-impl FromStr for Theme {
-    type Err = String;
+impl Theme {
+    pub fn from_name(name: &str) -> Option<Theme> {
+        match name {
+            "slick" => Some(SLICK.clone()),
+            "solarized-dark" => Some(SOLARIZED_DARK.clone()),
+            "solarized-light" => Some(SOLARIZED_LIGHT.clone()),
+            "plain" => Some(PLAIN.clone()),
+            "modern" => Some(MODERN.clone()),
+            "bad-wolf" => Some(BAD_WOLF.clone()),
+            "gruvbox-light" => Some(GRUVBOX_LIGHT.clone()),
+            "gruvbox-dark" => Some(GRUVBOX_DARK.clone()),
+            "space-villain" => Some(SPACE_VILLAIN.clone()),
+            _ => None,
+        }
+    }
+
+    pub fn from_file(file: &str) -> Option<Theme> {
+        let full_path = Path::new(file);
+        let xdg_path = util::xdg_config_home()
+            .join("i3status-rs/themes")
```

Comment:
We should be consistent with the existing config location here and use `i3status-rust/themes` (here an elsewhere).

Reply:
Thanks for your feedback, I'll fix that.